pkg/assets/store.go

```diff
@@ -201,19 +201,43 @@ func (s *Store) AddOAuth2(ctx context.Context, ns string, oauth2 *monitoringv1.O
 	return nil
 }
 
-// AddBearerToken processes the given SecretKeySelector and adds the referenced data to the store.
-func (s *Store) AddBearerToken(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error {
+// AddToken processes the given SecretKeySelector and adds the referenced data to the store.
+func (s *Store) addToken(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error {
 	if sel.Name == "" {
 		return nil
 	}
 
-	bearerToken, err := s.GetSecretKey(ctx, ns, sel)
+	token, err := s.GetSecretKey(ctx, ns, sel)
+	if err != nil {
+		return errors.Wrap(err, "failed to get token from secret")
+	}
+
+	s.TokenAssets[key] = Token(token)
+
+	return nil
+}
+
+func (s *Store) AddBearerToken(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error {
+	err := s.addToken(ctx, ns, sel, key)
 	if err != nil {
 		return errors.Wrap(err, "failed to get bearer token")
 	}
+	return nil
+}
 
-	s.BearerTokenAssets[key] = BearerToken(bearerToken)
+func (s *Store) AddAuthorizationCredentials(ctx context.Context, namespace string, auth *monitoringv1.Authorization, key string) error {
+	if auth == nil {
+		return nil
+	}
 
+	if err := auth.Validate(); err != nil {
+		return err
+	}
+
+	err := s.addToken(ctx, namespace, *auth.Credentials, key)
```

Comment:
Will this trigger a null pointer exception if `Credentials` is not set?

Reply:
It sure would, thanks! Added a guard and test case.